src/hash_table.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+from typing import Union
+import random
+
+
+class Node:
+    def __init__(self, key: str, val: Union[str, float], next: Node = None):
+        self.key = key
+        self.val = val
+        self.next = next
+
+
+class ChainHashTable:
+    def __init__(self, n: int):
```

Comment:
Parameter name should be `capacity` instead of `n`. We should also have default value for capacity in case user doesn't want to give a capacity.